### ISSUE TITLE
Question about the lengths of reasoning tokens

### ISSUE THREAD

**Issue by huhk-sysu** [UCSC-VLAA/m1#3]
In Figure 1, accuracy improves as thinking length increases up to ~4k tokens, whereas Figure 3 shows that increasing length by allowing multiple reasoning times in a budget forcing manner does not enhance accuracy.

Does this suggest that, in an unconstrained setting, the model naturally tends toward ~4k reasoning tokens? For example, if reasoning were typically shorter (e.g., 2k tokens), we would expect Figure 3 to show accuracy gains when increasing the budget focing times from 0 to 1 (since 2k ~ 4k falls within the improving regime of Figure 1). 

However, Figure 6 shows that the token lengths of the training set is much shorter, most of them are less than 2k tokens. Thus it's unlikely for the triained model to generate 4k reasoning tokens without budget forcing. This makes me confused.

Please tell me if I misunderstood, thank you!

**Comment by xk-huang** [comment]
Hi @huhk-sysu ,

Thanks for your interest in our work! 

Currently, we only apply "budget", aka the maximum number of thinking tokens limit, when generating answers, which means the model may not adhere to the budget and stop earlier (not generating enough tokens). As we found, the 4K budget is enough for a reasonably good performance. 

Directly forcing the model to "think" more like s1 won't work for medical questions, usually due to not enough medical knowledge inside the model (see "failure case analysis"). No matter how hard the model thinks, the missing knowledge cannot be come up.

As for the question:
> However, Figure 6 shows that the token lengths of the training set is much shorter, most of them are less than 2k tokens. Thus it's unlikely for the triained model to generate 4k reasoning tokens without budget forcing. 

Figure 6 shows the distribution of token lengths for questions. Although in the **training set**, a large number of questions can be solved within 2k tokens, there are still "long-tail" questions that require more tokens. Our finding shows that such observations hold still in the test set.


Besides, since we follow the method of s1, the thinking process is either truncated (reaching the budget) or stops early. There is still a large room to improve the precision of budget controlling when doing test-time scaling! Hoping to push the frontier a little more!
